napari/utils/events/containers/_evented_list.py

```diff
@@ -119,6 +118,10 @@ def __setitem__(self, key, value):
                         deferred=True,
                     )
                 )
+            if all(
+                new_el is old_el for new_el, old_el in zip_longest(value, old)
+            ):
+                return
```

Comment:
value could be any iterable. especially this check could consume this iterable. I think that you should create a list from the value variable. (then check could be simplified to list compare)